src/Readers/Vipr.Reader.OData.v4/OdcmReader.cs

```diff
@@ -518,47 +532,76 @@ private bool IsOperationBoundTo(IEdmOperation operation, IEdmEntityType entityTy
+                    odcmProperty.Projection = new OdcmProjection
+                    {
+                        Type = odcmType,
+                        BackLink = odcmProperty
+                    };
 
-                AddVocabularyAnnotations(odcmProperty, entitySet);
+                    try
+                    {
+                        _propertyCapabilitiesCache.Add(odcmProperty, OdcmCapability.DefaultEntitySetCapabilities);
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.Error(e, e.Message);
+                    }
+
+                    AddVocabularyAnnotations(odcmProperty, entitySet);
 
-                odcmClass.Properties.Add(odcmProperty);
+                    odcmClass.Properties.Add(odcmProperty);
+                }
+                // If we hit an invalid type, skip this property
+                catch (Exception e)
+                {
+                    Logger.Error(e, e.Message);
+                }
             }
 
             private void WriteProperty(OdcmClass odcmClass, IEdmSingleton singleton)
             {
-                var odcmType = ResolveType(singleton.EntityType().Name, singleton.EntityType().Namespace);
-                var odcmProperty = new OdcmSingleton(singleton.Name)
+                try
                 {
-                    Class = odcmClass,
-                    IsLink = true
-                };
+                    var odcmType = ResolveType(singleton.EntityType().Name, singleton.EntityType().Namespace);
+                    var odcmProperty = new OdcmSingleton(singleton.Name)
+                    {
+                        Class = odcmClass,
+                        IsLink = true
+                    };
 
-                odcmProperty.Projection = new OdcmProjection
-                {
-                    Type = odcmType,
-                    BackLink = odcmProperty
-                };
+                    odcmProperty.Projection = new OdcmProjection
+                    {
+                        Type = odcmType,
+                        BackLink = odcmProperty
+                    };
 
-                _propertyCapabilitiesCache.Add(odcmProperty, OdcmCapability.DefaultSingletonCapabilities);
+                    try
+                    {
+                        _propertyCapabilitiesCache.Add(odcmProperty, OdcmCapability.DefaultSingletonCapabilities);
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.Error(e, e.Message);
+                    }
 
-                AddVocabularyAnnotations(odcmProperty, singleton);
+                    AddVocabularyAnnotations(odcmProperty, singleton);
 
-                odcmClass.Properties.Add(odcmProperty);
+                    odcmClass.Properties.Add(odcmProperty);
+                }
+                catch (Exception e)

```

Comment:
We should add comments here around what exceptions we expect. Since we don't have explicit exception types, code readers will have a better expectations. At some point, we can move to collecting specific exception types and handling them differently. Same comment for all other {catch} blocks.